lib/disposable.dart

```diff
@@ -37,6 +37,43 @@ class _InternalDisposable implements _Disposable {
   }
 }
 
+/// A [Timer] implementation that exposes a [Future] that resolves when a
+/// non-periodic timer finishes it's callback or when any type of [Timer] is
+/// cancelled.
+class _ObservableTimer implements Timer {
+  Completer<Null> _didConclude = new Completer<Null>();
+  Timer _timer;
+
+  _ObservableTimer(Duration duration, void callback()) {
+    _timer = new Timer(duration, () {
+      callback();
+
+      if (!_didConclude.isCompleted) {
```

Comment:
This guard is unnecessary in reality. The underlying `Timer` implementation would never call this function if it were cancelled.